eth/handler_eth.go

```diff
@@ -34,7 +34,24 @@ import (
 type ethHandler handler
 
 func (h *ethHandler) Chain() *core.BlockChain { return h.chain }
-func (h *ethHandler) TxPool() eth.TxPool      { return h.txpool }
+
+// NilPool satisfies the TxPool interface but does not return any tx in the
+// pool. It is used to disable transaction gossip.
+type NilPool struct {
+}
+
+// NilPool Get always returns nil
+func (n NilPool) Get(hash common.Hash) *types.Transaction {
+	return nil
+}
+
+func (h *ethHandler) TxPool() eth.TxPool {
+	if h.disableTxGossip {
+		return &NilPool{}
+	} else {
+		return h.txpool
+	}
```

Comment:
```suggestion
	}
	return h.txpool
```
Sorry for the nit, just nice to have a small diff :sweat_smile:

Reply:
No worries, will update

Reply:
Done.